R/estimateParams.R

```diff
@@ -53,3 +53,366 @@ prepareData <- function(data) {
+  params <- c(unlist(transition$hazards, use.names = FALSE), unlist(transition$weibull_rates, use.names = FALSE))
+  exp(-params[trans] * t^params[trans + 3])
+}
+
+#' Retrieve Initial Parameter Vectors for Likelihood Maximization
+#'
+#' @param transition (`ExponentialTransition` or `WeibullTransition`)\cr containing the initial parameters.
+#'   See [exponential_transition()] or [weibull_transition()] for details.
+#'
+#' @return Returns a vector (`numeric`) of initial parameters for likelihood maximization.
+#' @export
+#'
+#' @details
+#' This function dispatches to either `getInit.ExponentialTransition` or `getInit.WeibullTransition`
+#' based on the `transition` object class.
+#'
+#' @examples
+#' transition <- exponential_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6)
+#' getInit(transition)
+getInit <- function(transition) {
+  assert_class(transition, "TransitionParameters")
+  UseMethod("getInit")
+}
+
+#' Retrieve Initial Parameters for Exponential Transition Model
+#'
+#' @param transition (`ExponentialTransition`)\cr containing the initial parameters.
+#'   See [exponential_transition()] for details.
+#'
+#' @return Returns a vector (`numeric`) of initial parameters for the exponential transition.
+#' @export
+#'
+#' @details
+#' Extracts initial parameter values specifically for an exponential transition model.
+#'
+#' @examples
+#' transition <- exponential_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6)
+#' getInit(transition)
+getInit.ExponentialTransition <- function(transition) {
+  unlist(transition$hazards, use.names = FALSE)
+}
+
+#' Retrieve Initial Parameters for Weibull Transition Model
+#'
+#' @param transition (`WeibullTransition`)\cr containing the initial parameters.
+#'   See [weibull_transition()] for details.
+#'
+#' @return Returns a vector (`numeric`) of initial parameters for the Weibull transition.
+#' @export
+#'
+#' @details
+#' Extracts initial parameter values specifically for a Weibull transition model.
+#'
+#' @examples
+#' transition <- weibull_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6, p01 = 2, p02 = 2.5, p12 = 3)
+#' getInit(transition)
+getInit.WeibullTransition <- function(transition) {
+  c(unlist(transition$hazards, use.names = FALSE), unlist(transition$weibull_rates, use.names = FALSE))
+}
+
+#' Generate the Target Function for Optimization
+#'
+#' @param params (`numeric` vector)\cr
+#'   Vector of parameters to be optimized over.
+#' @param data (`data.frame`)\cr
+#'   Data frame in the format created by [prepareData()].
+#' @param transition (`transitionParameters` object)\cr
+#'   Object specifying the distribution family. See [exponential_transition()] or [weibull_transition()] for details.
+#'
+#' @return Returns a function that calculates the negative log-likelihood (`numeric`) for the given parameters.
+#' @export
+#'
+#' @details
+#' This function creates a target function for optimization, which computes the negative log-likelihood for given
```

Comment:
at the moment it does not create a function yet. But I agree it would be better if it creates a function. Otherwise we do the if/else thing every time this is called.